[DEVTOOLS-485] WordPress Project Setup for Forum One & Pantheon Hosting

timnolte opened the pull request:
* Adds Buildkite pipeline configuration for Forum One & Pantheon hosting.
* Adds Buildkite Docker Compose configuration for builds.
* Updates Composer configuration with base set of starter plugins & package dependencies.
* Adds code quality linting support for PHP Code Sniffer & PHP Mess Detector.
* Adds code formatting EditorConfig configuration.
* Adds Capistrano build configuration for Forum One hosting.
* Adds Pantheon hosting support configuration.
* Adds DDev starting configuration.
  * Adds custom Gesso theme service and supporting commands.
  * Adds custom Redis service and support commands.
  * Adds code quality check commands.
  * Adds Pantheon hosting provider support configuration.
* Adds starter webroot directory with WP-CFM bundles & ACF JSON configuration support.
* Adds universal hosting support WP Starter setup for WordPress configuration.
* Adds universal hosting support for Must Use Plugins for ACF & WP-CFM.
* Updates `.gitignore` & `.dockerignore` for DDev & Buildkite builds.
* Changes repository `README.md` to be the build documentation with instruction for Forum One and Pantheon hosting.
* Moves the project `README.md` template to `README.project.md`
* Adds a default `robots.txt` for a WordPress site.

slucero commented on .buildkite/pipeline.f1.yml:
```diff
@@ -0,0 +1,156 @@
+      - forumone/capistrano:
+          require-stage: true
+          branches:
+            prerelease: prerelease
+            release: release
```
These branches aren't included in the step filters so they'll never be deployed with Capistrano.

slucero commented on .buildkite/pipeline.pantheon.yml:
```diff
@@ -0,0 +1,178 @@
+          from: /var/www/html
+          to: ./
+      # Push artifacts to Git remote.
+      - forumone/artifact-push#v0.3.0:
```
```suggestion
      - forumone/artifact-push#v0.3.1:
```

slucero commented on Dockerfile:
```diff
@@ -69,7 +69,7 @@ FROM release AS test
 COPY --from=composer-dev ["/app/web", "web"]
 
-COPY --from=gesso-dev ["/app", "web/wp-content/themes/gesso"]
+COPY --from=gesso-dev ["/app", "web  /wp-content/themes/gesso"]
```
```suggestion
COPY --from=gesso-dev ["/app", "web/wp-content/themes/gesso"]
```

slucero commented on README.md:
```diff
@@ -1,62 +1,200 @@
+    * `.ddev/providers/pantheon.yaml` > `[PROJECT.KEY]`, see the instructions in the `pantheon.yml` file.
+* Delete `.buildkite/pipeline.f1.yml`.
+* Delete the `web/wp-content/config` directory.
+* Delete the `capistrano` directory.
```
```suggestion
* Delete the `capistrano` directory.
* Delete the `Capfile` file
* Delete the `Gemfile` file
```

While adding this, I also noticed the `Gemfile` isn't present at the root of the repo which I believe will cause the Capistrano steps to fail.

timnolte commented:
@slucero requested changes made. I also added 3 additional Pantheon specific WordPress plugins that we should be using and updated the README to speak to those in terms of removing for F1 Hosting. Thanks!
